sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveExternalCatalogSuite.scala

```diff
@@ -218,4 +218,29 @@ class HiveExternalCatalogSuite extends ExternalCatalogSuite {
     val alteredTable = externalCatalog.getTable("db1", "parq_tbl")
     assert(alteredTable.provider === Some("foo"))
   }
+
+  test("SPARK-31751: serde property `path` overwrites hive table property location") {
+    val catalog = newBasicCatalog()
+    val hiveTable = CatalogTable(
+      identifier = TableIdentifier("parq_alter", Some("db1")),
+      tableType = CatalogTableType.MANAGED,
+      storage = storageFormat,
+      schema = new StructType().add("col1", "int"),
+      provider = Some("orc"))
```

Comment:
Conventionally, we use `provider = Some("parquet"))` for the tables who names start with `parq_`.